go-controller/pkg/ovn/master.go

```diff
@@ -831,6 +850,13 @@ func (oc *DefaultNetworkController) deleteNodeEvent(node *kapi.Node) error {
 	if err := oc.deleteNode(node.Name); err != nil {
 		return err
 	}
+
+	if chassisID, err := util.ParseNodeChassisIDAnnotation(node); err == nil {
+		if err = libovsdbops.DeleteChassisTemplateRecord(oc.nbClient, []*nbdb.ChassisTemplateVar{{Chassis: chassisID}}...); err != nil {
+			return fmt.Errorf("failed deleting chassis template variables for %s: %v", node.Name, err)
+		}
+	}
```

Comment:
Why not do this inside `deleteNode` where the DeleteChassis is also done?

Reply:
We need the chassisID and we don't have that in deleteNode.  I can move the annotation parsing and the deletion there if it looks better to you.

Reply:
Well, I am bit confused here.

Within `deleteNode` we are deleting the chassis as well. So I thought that the chassisID could somewhat be related to the chassis we are deleting as well and that could be re-used for both things. But I see that we are doing that with a hostname predicate instead and that chassisID is not an index on the chassis sbdb table.

I still think though that both operation are indirectly related and could make sense having together (or does the chassis ID have any meaning when the chassis does not exists or something?)

Could you also add this previous answer you gave me as a comment on why we are ignoring the error (it asked myself the same question again just to realize I had asked it before)
```
Some nodes might not have the chassisID annotation (yet). That's should not be a problem. Either they don't need one or it wasn't there yet before the node got deleted. In any case, we have not created a chassis template record for that node.
```

Reply:
Chassis deletion and Operations are indirectly related but the problem is NB.Chassis_Template_Var identifies the chassis by name, not by hostname.  The original OVN patch tried to match the template_var.chassis against both a chassis ID OR a chassis hostname but had issues and we had to go for the version that matches only against chassis ID.

I'll add a comment and I'll also mention the part you pasted above for the ignored error.